### ISSUE TITLE
The first dimension of input and shift should be equal in tin_shift op, or will cause error of crossing the border.

### ISSUE THREAD

**Issue by Mrxiaofei** [open-mmlab/mmcv#1703]
Thanks for reporting the unexpected results and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmcv.readthedocs.io/en/latest/trouble_shooting.html) but cannot get the expected help.
3. The unexpected results still exist in the latest version.

**Describe the Issue**
The first dimension of input and shift should be equal in tin_shift op. If not, mat cause the error of crossing the border.
In [tin_shift_cuda_kernel.cuh](https://github.com/open-mmlab/mmcv/blob/master/mmcv/ops/csrc/common/cuda/tin_shift_cuda_kernel.cuh) line 20, 'const int n_index = (index / hw_size / channels) % batch_size;', if batch_size which is input batch_size is larger than batch_size of shift, the sentence 'int t_shift = shift[n_index * group_size + group_id];' will cause the error of crossing the border. But in cuda the error of crossing the border will not warn the user.

**Reproduction**

1. What command, code, or script did you run?

```bash
A placeholder for the command.
```

2. Did you make any modifications on the code? Did you understand what you have modified?

**Environment**

1. Please run `python -c "from mmcv.utils import collect_env; print(collect_env())"` to collect necessary environment information and paste it here.
2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
If applicable, paste the error traceback here.

```none
A placeholder for traceback.
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by grimoire** [comment]
Yes! You are right, batch_size should be the same between `input` and `shift`. We have mentioned it in [docsting](https://github.com/open-mmlab/mmcv/blob/26aba2f533507f6e4568d182702eca52922b9649/mmcv/ops/tin_shift.py#L64) of `tin_shift` in python. 
I think it is a good idea to add an assertion or warning in python to avoid batch_size mismatch. You are welcome to send a PR to fix it if you want.
